hibernate-core/src/main/java/org/hibernate/boot/Metadata.java

```diff
@@ -203,4 +205,12 @@ public interface Metadata extends Mapping {
 	 * All of the known model contributors
 	 */
 	Set<String> getContributors();
+
+	TypeConfiguration getTypeConfiguration();
+
+	Type getIdentifierType(String className) throws org.hibernate.MappingException;
+
+	String getIdentifierPropertyName(String className) throws org.hibernate.MappingException;
+
+	Type getReferencedPropertyType(String className, String propertyName) throws MappingException;
```

Comment:
I wonder if we should just make these default methods, delegating to `#getTypeConfiguration`?

```suggestion
	default Type getIdentifierType(String className) throws org.hibernate.MappingException {
		return getTypeConfiguration().getIdentifierType( className );
	}

	String getIdentifierPropertyName(String className) throws org.hibernate.MappingException {
		return getTypeConfiguration().getIdentifierPropertyName( className );
	}

	Type getReferencedPropertyType(String className, String propertyName) throws MappingException{
		return getTypeConfiguration().getReferencedPropertyType( className, propertyName );
	}
```